lib/pymedphys/_experimental/streamlit/apps/iview_to_dicom.py

```diff
@@ -72,19 +144,60 @@ def _create_portal_image_dicom_dataset(
 
     pixel_spacing = 1 / pixels_per_mm * sid / sad
 
+    date = datetime.strftime("%Y%m%d")
+    time = datetime.strftime("%H%M%S.%f")
+
+    # Many of the below was input without a full understanding of the
+    # implications of the choices being made.
     ds = _pp_dcm_create.dicom_dataset_from_dict(
         {
-            "ImageType": ["ORIGINAL", "PRIMARY", "PORTAL"],
+            "Modality": "RTIMAGE",
+            "InstanceCreatorUID": PYMEDPHYS_ROOT_UID,
+            "SOPClassUID": "1.2.840.10008.5.1.4.1.1.481.1",
+            "SOPInstanceUID": pydicom.uid.generate_uid(prefix=f"{PYMEDPHYS_ROOT_UID}."),
+            # From <https://dicom.innolitics.com/ciods/cr-image/general-image/00080008>
+            #
+            # * is the image an ORIGINAL Image; an image whose pixel
+            #   values are based on original or source data
+            # * is the image a SECONDARY Image; an image created after
+            #   the initial patient examination
+            "ImageType": ["ORIGINAL", "SECONDARY", "PORTAL"],
+            # Purposefully avoided all concepts of 'Study' if possible
+            # as this would require appropriately grouping the images
+            # from each field. Certainly doable, and arguably it should
+            # be done, but out of scope for the current task.
+            "PatientID": patient_id,
+            "PatientName": patient_name,
+            "AcquisitionDate": date,
+            "ContentDate": date,
+            "AcquisitionTime": time,
+            "ContentTime": time,
             "Rows": pixel_array.shape[0],
             "Columns": pixel_array.shape[1],
             "GantryAngle": gantry_angle,
-            "BeamLimitingDeviceAngle ": collimator_angle,
-            "PatientSupportAngle ": table_angle,
-            "PixelData": pixel_array,
+            # Not correct, there may have been many images in this acquisition
+            "ImagesInAcquisition": 1,
+            "PatientOrientation": "",
+            "SamplesPerPixel": 1,
+            "PhotometricInterpretation": "MONOCHROME2",
+            "BeamLimitingDeviceAngle": collimator_angle,
+            # Can be implemented in general
+            "BitsAllocated": 16,
+            "BitsStored": 16,
+            "HighBit": 15,
+            "RTImageLabel": "",
+            "RTImagePlane": "NORMAL",
+            "PixelRepresentation": 0,
+            "XRayImageReceptorAngle": 0.0,
+            "RTImagePosition": None,
+            "PatientSupportAngle": table_angle,
+            "PixelData": pixel_array.tobytes(),
             "RadiationMachineSAD": sad,
             "RTImageSID": sid,
+            "PrimaryDosimeterUnit": "",
```

Comment:
MU should be fine.
If they are KV images, I'll need to look it up, but having something might be better than empty, and MU covers the vast majority of situations and systems

Reply:
For this one, I'd prefer to err on the side of agreeing with what is currently exported from the iView database by the official tool, which, in this case is `""`.

Reply:
> For this one, I'd prefer to err on the side of agreeing with what is currently exported from the iView database by the official tool, which, in this case is `""`.

I will submit a defect to the vendor

Reply:
🙂